WIP: make all scripts py3 compatible

m4rc1e opened the pull request:
I'll make sure they're still compatible with py2... even though it will be unsupported soon.

davelab6 commented:
Is this ready to be merged?

felipesanches commented:
I would like to merge this after reviewing the code and also ensuring that Travis is properly running in this repo. We do have a `.travis.yml` in the repo, but it seems that someone with the proper access rights (maybe @davelab6 ?) still needs to enter the Travis setup panel and enable it for `googlefonts/gftools`. I am a bit surprised that we still don't have that enabled, so maybe there's some other reason for the Travis builds not being triggered here.

anthrotype commented:
I just enabled it https://travis-ci.org/googlefonts/gftools
try to push an empty commit to trigger the build

anthrotype commented on setup.py:
```diff
@@ -73,6 +73,7 @@ def gftools_scripts():
         'protobuf',
         'requests',
         'tabulate',
+        'fs',
```
if you don't import directly `fs` (aka pyfilesystem2), but you included it only to be able to import fontTools.ufoLib, I would recommend that you specify the fonttools requirement using the appropriate 'extra' notation, i.e. `fonttools[ufo]`, instead of only `fonttools`. This will make sure that the required extra requirements are met (including fs).

m4rc1e commented:
I think this got blocked in an email and @anthrotype was supposed to take a look. I still think there are a few script which need converting. Leave for now.

felipesanches commented:
@m4rc1e could you please `rebase & push --force` here to trigger a Travis build, please?

m4rc1e commented:
Done.

It will be great to get this finished at some point very soon.

felipesanches commented:
@marcie, from Travis build log:
```
  File "/home/travis/virtualenv/python3.7.1/bin/gftools-add-font.py", line 85, in _FileFamilyStyleWeights
    raise OSError(errno.ENOTDIR, 'No such directory', fontdir)
NotADirectoryError: [Errno 20] No such directory: './data/mock_googlefonts/ofl/abel'
The command "sh run.sh" exited with 1.
```

m4rc1e commented:
@felipesanches I'll sort this now. This quarter @anthrotype is working on updating the gf engs python tools to py3. We should leave this open until he has had a look and everything is in sync.

m4rc1e commented:
@felipesanches I'll tidyup this up next week. I need to fix the tests so travis doesn't complain. Commits are also pretty messy.

felipesanches commented:
thanks, @m4rc1e

m4rc1e commented:
Just found this in my mails from Rod

> Cosimo is assigned migrate internal code, after which the public stuff can be updated. Please don't do it independently, though if Cosimo wants to collaborate on it that's workable. It causes trouble when the public copies drift from the internal ones.

Keep this open until Cosimo is happy. In the meantime, I'll make sure this pr passes travis.

felipesanches commented:
thanks, @m4rc1e

m4rc1e commented:
I had a chat a month back with @anthrotype. He says this repo has diverged too much when compared against the internal version. I think it's best I finish off this pr. Once merged I can compare this version against the internal and try to resync them.

davelab6 commented:
>  I had a chat a month back with @anthrotype. He says this repo has diverged too much when compared against the internal version. I think it's best I finish off this pr. Once merged I can compare this version against the internal and try to resync them.

I'm curious about the state of this, @anthrotype and @m4rc1e  :)

m4rc1e commented:
@davelab6 Pretty much wrapped up. I've still got to make a test pypi package to see if it is pip installable.

chrissimpkins commented:
I recognize that this is not a full transition to Py3 only support and I don't know how common it is to execute modules with direct paths but FWIW...

As of macOS 10.15.1 and my install of Ubuntu 19.10 `/usr/bin/env python` is Python 2.7.17.  Should the shebang in the modules be the following?

```python
#!/usr/bin/env python3
```

I think that we have to assume that any module included here must be Py3.6+ only if the project is based on an unpinned version of fontTools because as of the v4.0.0 release there is no longer a Py2 support guarantee. 

Also, Niko has recently informed me that the `# coding: utf-8` line is not necessary when/if the transition to Py 3 only support occurs.

m4rc1e commented:
I'm done. I'll make a test pypi package and see if I can pip install it.

@chrissimpkins thank you for your suggestions. I've done some reading and I agree that the python3 shebang is needed and we can also drop the encoding line.

m4rc1e commented:
Package is working nicely on test pypi.

I've also bumped the version number to 0.30. We should make it 1.0.0, once we've removed redundant scripts and added further polish.

@felipesanches this is ready for review if you have the time? If you'd like to install the test pypi package, do

```
pip install -i https://test.pypi.org/simple/ gftools==0.3.0 --extra-index-url https://pypi.org/simple --no-cache-dir
```

It would be great to get this approved sometime this week so we're py3 before the py2 EOL.

Another big win is the CI tests have been fixed during the conversion.
